docs/deprecation-guide.md

```diff
@@ -0,0 +1,139 @@
+# When and how to deprecate hooks and non-private functions in subscriptions-core
+
+When breaking changes are made to a filter, action or non-private function in subscriptions-core, we need to continue to support the current behaviour of it while encouraging downstream extensions or themes to use the updated version.
+
+When we mark a function or hook as deprecated, a deprecation notice warning will display when the function or hook is used.
+
+-   [Filters](#filters)
+-   [Actions](#actions)
+-   [Non-private functions](#non-private-functions)
+
+## Identifying functions needing deprecation
```

Comment:
This looks like a good working definition of "breaking changes" – would be good to link to the summary above (e.g. `How do we identify breaking changes` or similar.